### ISSUE TITLE
Custom scalars can't be made typesafe

### ISSUE THREAD

**Issue by Luka4ever** [graphql-nexus/schema#360]
It's currently not possible to make custom scalars typesafe since the generated type for fields with a custom scalar type is `any`, an example of this can be seen [here](https://github.com/prisma-labs/nexus/blob/develop/examples/kitchen-sink/src/kitchen-sink-typegen.ts#L74). I'm not sure if this is simply a bug, or if the feature hasn't been implemented yet.

**Comment by santialbo** [comment]
Have a look at this other example. https://github.com/prisma-labs/nexus/blob/bd452ebbe650e1cfa2cbb562c26ccbffd2bd7fca/examples/ghost/src/ghost-schema.ts

By providing `typegenAutoConfig.backingTypeMap` on the schema creation you are able to define the type of custom scalars.

```ts
import { GraphQLDateTime } from "graphql-iso-date";
import { asNexusMethod } from "nexus";

export const DateTime = asNexusMethod(GraphQLDateTime, "datetime");
```

```ts
  ...
  typegenAutoConfig: {
    ...
    backingTypeMap: {
      DateTime: "Date"
    }
  }
```

**Comment by Luka4ever** [comment]
Thank you, I must have missed that example when I was reading through the repo. Perhaps some documentation should be added about how to implement custom scalars.

**Comment by jasonkuhrt** [comment]
Resolved by #469 I think. Thanks @Sytten
